core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -176,7 +192,17 @@ public void validateIndexName(String index, ClusterState state) {
         }
     }
 
-    public void createIndex(final CreateIndexClusterStateUpdateRequest request, final ActionListener<ClusterStateUpdateResponse> listener) {
+    public void createIndex(final CreateIndexClusterStateUpdateRequest request,
+                            final ActionListener<ClusterStateUpdateResponse> listener) {
+
+        final int numShardCopies = IndexMetaData.INDEX_NUMBER_OF_REPLICAS_SETTING.get(request.settings()) + 1;
+        if (request.waitForActiveShards().exceeds(numShardCopies)) {
```

Comment:
Actually, I'm able to easily keep the check while separating the waiting part out into the `TransportCreateIndexAction`.  It does make the code simpler and less overall changes.  The question is, should `TransportRolloverAction` wait for active shards just like `TransportCreateIndexAction`?